File: golang/0011.go
func maxArea(height []int) int {
	max := 0
	left := 0
	right := len(height) - 1

	for left < right {
		width := right - left
		max = Max(max, Min(height[left], height[right])*width)

		if height[left] < height[right] {
			left++
		} else {
			right--
		}
	}
	return max
}

func Min(x, y int) int {
	if x < y {
		return x
	}
	return y
}

func Max(x, y int) int {
	if x < y {
		return y
	}
	return x
}

File: rust/0011.rs
impl Solution {
    pub fn max_area(height: Vec<i32>) -> i32 {
        let mut max = 0;
        let mut left = 0;
        let mut right = height.len() - 1;

        while left < right {
            let width = right - left;
            let mut h = height[left];
            if height[right] < height[left] {
                h = height[right];
                right -=1;
            } else {
                left += 1;
            }
            let area = width * h as usize;
            if area > max {
                max = area
            }
        }
        max as i32
    }
}
